compiler/tests/org/jetbrains/kotlin/cli/LauncherScriptTest.kt

```diff
@@ -359,4 +360,70 @@ fun f() : Result<Int> = Result.success(42)
             expectedExitCode = 1
         )
     }
+
+    fun testRunClassFileWithExtension() {
+        runProcess("kotlinc", "$testDataDirectory/helloWorld.kt", "-d", tmpdir.path)
+        runProcess(
+            "kotlin",
+            "test/HelloWorldKt.class",
+            expectedStdout = "Hello!\n",
+            workDirectory = tmpdir
+        )
+        runProcess(
+            "kotlin",
+            "test.HelloWorldKt.class",
+            expectedExitCode = 1,
+            expectedStderr = "error: could not find or load main class test.HelloWorldKt.class\n",
+            workDirectory = tmpdir
+        )
+
+        val subDir2 = File("$tmpdir/subDir1/subDir2").also { it.mkdirs() }
+        val subDir3 = File("$tmpdir/subDir3").also { it.mkdir() }
+        val subDir1 = subDir2.parentFile
+        fun makeSureTheClassFileExists(classFile: File) {
+            if (!classFile.exists()) {
+                throw FileNotFoundException("class file $classFile is not found")
```

Comment:
Minor, but let's throw `AssertionError` to make it clear that the test fails because of incorrect compiler behavior, not some I/O problems